templates/_components/_image.twig

```diff
@@ -2,14 +2,266 @@
+  @param   alt                  {string}                  Alt attribute for the <img> tag
+  @param   background           {string}                  Class name for background image
+  @param   lazyload             {bool}                    Whether to lazyload images (default: true)
+  @param   webp                 {bool}                    Whether to generate WebP versions (default: true)
+
+  Example of simple usage
+  -----------------------
+  ```
+  <figure>
+    {% include '_components/_image' with { params: {
+      asset: image,
+      class: 'profile-pic',
+      preset: 'avatar',
+    } } %}
+  </figure>
+  ```
+
+  Example of usage with breakpoint asset
+  --------------------------------------
+  ```
+  <figure>
+  {% include '_components/_image' with { params: {
+    asset: image,
+    breakpoint_assets: {
+      '--lg-plus': imageLarge,
+      '--md-plus': imageMedium,
+      '--sm-plus': imageSmall,
+    },
+  } } %}
+  </figure>
+  ```
+
+  Set `presets`(a set of image transformation configurations) variable
+  values in `_image` component. Define the desired preset values in the
+  image component as shown in the examples below.
+
+  Preset options
+  --------------
+  @option   media          {string}      Any valid media query that will be used in `media` attribute in the `<img>` tag
+  @option   breakpoints    {array}       Transform values for different view breakpoints that will generate `source` tags
+  @option   sizes          {array}       Image size values for different images to render in `srcset` attribute of the `<img>` tag
+  @option   transform      {array}       Aspect and position related transform values
+  @option   guide          {string}      Any valid media query that will be used in `sizes` attribute in the `<img>` tag
+
+  Set `queries` variable to an array of valid media queries. It will be
+  used to define different display device sizes. 
+
+  Queries examples
+  ---------------
+
+  1)  Query used for `<img>` tag `media` attribute
+      ```
+      {% set queries = {
+        '--xs-plus': '(min-width: 448px)',
+        '--sm-plus': '(min-width: 900px)',
+        '--md-plus': '(min-width: 1200px)',
+      } %}
+      {% set presets = {
+        blog: {
+          breakpoints: [{
+            media: '--md-plus',
+            sizes: [
+              { width: 500 },
+              { width: 600 },
+              { width: 700 },
+              { width: 800 },
+            ],
+            transform: { mode: 'crop', ratio: 0.9 },
+          },{
+            sizes: [
+              { width: 400 },
+              { width: 600 },
+              { width: 900 },
+            ],
+            transform: { mode: 'crop', ratio: 1.1 },
+          },
+        ]}
+      } %}
+      ```
+      
+      HTML Output:
+      ```
+      <picture>
+        <source
+          width="800"
+          height="889"
+          media="(min-width: 1200px)"
+          sizes="576px"
```

Comment:
Why is the value of `sizes` set to `567px` in this case? We did not specify a `guide` param in our preset, did we?

Reply:
Removed `sizes`.